### PR TITLE
asm/c64:  Adding asm dot functions

### PR DESCRIPTION
**Pull request by Kunde21**
Migrated from https://github.com/gonum/internal/pull/57

### CODE REVIEW

File: internal/asm/c64/benchDot_test.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	}{
+		{"DotcUnitary", DotcUnitary},
+		{"DotuUnitary", DotuUnitary},
+	} {
+		for _, v := range []int64{1, 2, 3, 4, 5, 10, 100, 1e3, 5e3, 1e4, 5e4} {
+			t.Run(fmt.Sprintf("%s-%d", tst.name, v), func(b *testing.B) {
+				x, y := x[:v], y[:v]
+				b.SetBytes(128 * v)
+				for i := 0; i < b.N; i++ {
+					_ = tst.f(x, y)
```
Comment (kortschak) [inline internal/asm/c64/benchDot_test.go]: Sorry, I missed this, assign to a sink.

File: internal/asm/c64/benchDot_test.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		for _, ln := range []int{1, 2, 3, 4, 5, 10, 100, 1e3, 5e3, 1e4, 5e4} {
+			for _, inc := range []int{1, 2, 4, 10, -1, -2, -4, -10} {
+				t.Run(fmt.Sprintf("%s-%d-inc%d", tst.name, ln, inc), func(b *testing.B) {
+					b.SetBytes(int64(128 * ln))
+					var idx int
+					if inc < 0 {
+						idx = (-ln + 1) * inc
+					}
+					for i := 0; i < b.N; i++ {
+						_ = tst.f(x, y, uintptr(ln), uintptr(inc), uintptr(inc), uintptr(idx), uintptr(idx))
```
Comment (kortschak) [inline internal/asm/c64/benchDot_test.go]: Here too.

**Comment by Kunde21** [comment]
PTAL 

Only change is variable renaming suggested in #207 and squashed into 66b4f42

**Comment by Kunde21** [comment]
Bump

ping @kortschak and @vladimir-ch

**Comment by kortschak** [comment]
I can take a look at these later today.

**Comment by kortschak** [comment]
I hate that github gets rid of objects that have been rebased away immediately.

**Pull request merged**